Added dogescript

ThomasHickman opened the pull request.

ThomasHickman committed:
Dne obv. ds stff

ThomasHickman committed:
Such tabs. mUch correct

ThomasHickman committed:
Much finish

ThomasHickman committed:
Debugged djs file

ThomasHickman committed:
Added dogescript compile on load

ThomasHickman committed:
Added webpack bundle to compile dogescript files

ThomasHickman committed:
Added powered by dogescript

GrahamCampbell commented on index.djs:
```diff
@@ -0,0 +1,75 @@
+wow& hideLoadingScreen()
+
+very animations = [/*
+*/    "spin",
```
don't need that comment anymore tbh

ThomasHickman replied:
The comment is needed because dogescript adds a semicolon to the end of every line starting with "very". I'll change this to statements constructing the array instead though e.g `animations dose push "zoom"` ...

ThomasHickman replied:
Done this in 9044cde6f513efb8693e52d7d8d9845d2ddad10d

GrahamCampbell commented:
We can't had a dependency on npm, because GitHub won't run that for us.

ThomasHickman commented:
The npm dependancy is only for showing how to compile the source file, all npm dependant things have been compiled using webpack into the included file - compileDJS.js

ThomasHickman committed:
Locked down the version of dogescript

ThomasHickman committed:
Contructed animations using statements

This is a much less hacky method of setting the value of animations
than using a javascript initialiser list

GrahamCampbell commented:
Server side/developer side, same problem. Can we avoid having to run tools?

ThomasHickman commented:
Only the dogescript compiler (and a small helper file) is being compiled using npm on the developer side, the dogescript source file is still being compiled on load by the compiled dogescript compiler - I don't think compiling dogescript on the developer side is going to be much of a problem. If it is, I could manually concatenate all files and dependencies of dogescript into one javascript file, but this is basically what webpack does in it's compilation process.
